cmd/sonobuoy/app/gen.go

```diff
@@ -128,14 +133,16 @@ func (g *genFlags) Config() (*client.GenConfig, error) {
 
 	return &client.GenConfig{
 		E2EConfig:            e2ecfg,
-		Config:               g.getConfig(),
+		Config:               g.resolveConfig(),
```

Comment:
Updated the name to try and be a bit more explicit that this mutates things. It was really awkward when I tried to break the function up. The default config, loaded config, e2e flags, mode, and plugin flags are all just intertwined. It would be nice to be more clear in the code, but I'll definitely settle for clarity via good testing.